### ISSUE TITLE
Create convenience method for web5.did.create('ion')

### ISSUE THREAD

**Issue by ALRubinger** [TBD54566975/web5-js#31]
Nothing broken here, but an opportunity for Web5 to erase complexity and be more approachable to newcomers. 

One of the first things we do in the QuickStart is have the user create a DID. When using the recommended:

```
let did = await web5.did.create('ion');
```

...we are left to immediately delve into conceptual topics that we think would make a stronger experience if we could avoid linking into those concepts:

* Why we pass a `String` in there
* What `ion` is
* What a DID Method Name is 

Is there some syntactic sugar we can add to the Web5 API which erases this? Perhaps:

* `web5.did.create();` // No-arg constructor which applies a default we choose, like `ion`?
* `web5.did.createIon();` // An Autocomplete version that tooling can help with so we don't need to pass a manual String?
* `web5.did.create(Web5.DID_METHODS.ION);` // A constructor that accepts a type and an enum of those types

Open to other ideas - whatever makes for easier autocomplete and intuitive API, and limits the conceptual weight of someone having to make a choice about DID method types, understanding why, and passing in an untyped String to designate which.

**Comment by frankhinek** [comment]
Hey @ALRubinger -- good question and callout.

My preference is:
- `web5.did.create();` // No-arg constructor which applies a default we choose, like ion?

I'd be ok with making the default `ion` and would be simple to implement.

@csuwildcat you cool with that?

**Comment by csuwildcat** [comment]
I personally would prefer not to have a default, in case we ever swap that default. This is a really impactful thing to default, and this would mean a significant breaking change for all downstream lib users if we ever did it.

**Comment by dcrousso** [comment]
i think at the very least we should at least have something like `web5.did.MethodName.Ion` so that developers don't have to know any magic strings (as well as having an easy way for the developer to log in the `console` what all the method names are)

having a default would be nice as it could let developers just DoTheThing™ `web5.did.create()` and not even have to worry about a method.

@csuwildcat i do get your concern about us being constrained in changing the default.  what if we had a `web5.did.DefaultMethodName` so that developers who do care about the default can know what it is and do logic based on it without having do awful stuff like looking at the source or checking the package version

**Comment by dcrousso** [comment]
i went ahead with the first part at least :) https://github.com/TBD54566975/web5-js/pull/32

**Comment by mistermoe** [comment]
@csuwildcat can you help me understand your decision making process for when using default args is acceptable? i ask because you use them quite a bit in [`ion-tools`](https://github.com/decentralized-identity/ion-tools#usage). Specifically:

```js
let authnKeys = await generateKeyPair();
```

[this defaults](https://github.com/decentralized-identity/ion-tools/blob/main/src/utils.js#L45-L59) to `secp256k1` but you can also provide `Ed25519`.

**Comment by csuwildcat** [comment]
Because it's unlikely we'd need to change that default, save a quantum computing break of EC keys, but DID Methods are more of a swappable thing. I'm fine with whatever you all want to do.

**Comment by mistermoe** [comment]
Only concern that comes to mind is: does a default DID method kinda feel like when browsers would automatically set themselves as your machine's default browser without asking you?

**Comment by csuwildcat** [comment]
Let's switch to did:key as the default and not require any args?

**Comment by ALRubinger** [comment]
`did:key` is great by me as a default. Lets you send messages, and removes conceptual weight for newcomers. When they wanna be addressable on the decentralized web, that's an appropriate time to intro them to `ion` and DID methods etc. Great way to move folks up the ladder.

@mistermoe wrote:
> Only concern that comes to mind is: does a default DID method kinda feel like when browsers would automatically set themselves as your machine's default browser without asking you?

Eh, not to me bud. The user can absolutely choose - they're choosing the simplest possible case which makes them not need to think about it. I equate it more to a command line that's like: "Enter the date (press [ENTER] to default to current date):" You can do anything but they give you an easy paved path to remove decision making and knowledge to go forward.

**Comment by ALRubinger** [comment]
@mistermoe What's a worse default - Can we make it so `web5-js` automatically installs an entire U2 album into your wallet? 🤘🏻

**Comment by dcrousso** [comment]
> Let's switch to did:key as the default and not require any args?

https://github.com/TBD54566975/web5-js/pull/33 :)

**Comment by frankhinek** [comment]
The only aspect about this default that is going to be odd is that outside of the initial introduction to this method, it is rather likely that service endpoints would need to be defined to include a DWN.  Since `did:key` doesn't support services, it will need to be an ION did.

I would think we'd want to make it as easy as possible for a dev to get started with a DID method that supports discovery and routing of messages to target DIDs.  Since that's ION for now, we may not want to completely hide the type of DID -- even if we make it simpler to use (e.g., with a symbol/enum instead of string).

**Comment by ALRubinger** [comment]
Bowing out of the part about "which DID method is the best default". Retracting my statement above about that being `did:key`. I know enough when to leave that to y;all :P My request principally - that we elect a default to slim first user experience. Thanks again for working through this.

**Comment by ALRubinger** [comment]
We're shipping the Quickstart without this in place; when done, please open an issue on the developer.tbd.website repo to update the Quickstart w/ the newer convenience method.

**Comment by ALRubinger** [comment]
@frankhinek @dcrousso Could you confirm whether this is in `0.6.0`, perhaps solved in separate issues? This is the one we're tracking to update docs, so we'll hold on this dep until resolved.

**Comment by frankhinek** [comment]
@ALRubinger after carefully considering the pros and cons and discussing with @csuwildcat and @mistermoe here is the current thinking:

**Selecting a Default**
- I won't rehash the debate on default DID methods.  We're using ION for a series of reasons until/unless another strong contender arises.  We want to encourage developers to select ION because `did:key` lacks basic security & endpoint publishing features.
    - Note: In case you are wondering,`did:key` support was added to Web5 JS primarily to facilitate the DID Connect protocol for applications to securely establish a channel with a remote agent/server hosting a DWN.  It is not intended that we encourage use of `did:key` for DWN tenants.
- Providing a default here means typing 5 less characters, but only for the initial getting started.  There will be developers new to the space don't understand what a "DID" much less "ion" but can copy & paste `web5.did.create()` or `web5.did.create('ion')` or `web5.did.createIon()` with the same effort.  Overtime, we can help them learn more about what these terms mean.
- One of the powerful things about ION DIDs is that you can specify service endpoints that allow for decentralized discovery, dereferencing, and routing using DIDs.  After the initial basic tutorials, the `web5.did.create()` calls will look more like the example below, and it seems unlikely anyone would do this and have no understanding about DIDs or DID methods.

    ```javascript
    const carolDid = await web5.did.create('ion', {
      services: [
         {
           'id': 'dwn',
           'type': 'DecentralizedWebNode',
           'serviceEndpoint': {
             'nodes': [
                'http://dwn.provider.com',
                'https://your-ngrok-subdomain.ngrok.io/dwn',
             ]
           }
         }
       ]
    });
    ```

**Enum vs. Symbol vs. String**
* It is unclear that either of these alternatives are objectively more intuitive and seem to be more of an individual style preference:
    * `web5.did.create(web5.did.DID_METHODS.ION)`
    * `web5.did.create('ion')`
* They both involve typing text into a text editor / IDE (symbol being more verbose), are subject to potential for mis-spelling, and both can be suggested as an auto-complete option by providing JSDoc (CJS) or TypeScript types.
* Additionally, the future goal is to provide flexibility so that developers can add support for other DID methods beyond those that may be pre-defined in the Web5 JS code base.  We'd like to enable simple registration of new DID method types based on the `"scheme:method"` (e.g., `"did:web"`).  This could be done with strings or symbols but the preference of the lib core maintainers is strings.

**What's wrong with strings?**
- There are plenty of examples of using strings through web standards and popular libraries.
- I won't attempt to produce a long list, but use of strings to specify configuration options and parameters when calling methods in JavaScript is widespread.
    - [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign) when specifying which `algorithm` to use.
    - Veramo [DID Manager](https://veramo.io/docs/api/core-types.ididmanager.didmanagercreate) and [Setup Guide](https://veramo.io/docs/node_tutorials/node_setup_identifiers)

In summary, we're going to leave the current method as is.  Since the DID Manager API is still evolving and there are many new capabilities to be added, we want to get those into the wild and get more feedback from developers.  If there is strong feedback that people are having issues with the use of strings to specify methods, algorithms, and related configuration parameters throughout Web5 JS we can revisit.

**Comment by frankhinek** [comment]
Two enhancements made in [v0.7.0](https://github.com/TBD54566975/web5-js/releases/tag/v0.7.0) that indirectly improve this situation so will close the issue:
- As a result of introducing the agent concept in Web5 JS, it is no longer necessary to create a DID just to get started.  It happens automatically
- While the command to create a DID is exactly the same as it was (`web5.did.create('ion')`) the 0.7.0 release involved conversion of the library to TypeScript as noted 3 weeks ago:
    > ...both can be suggested as an auto-complete option by providing JSDoc (CJS) or TypeScript types
